src/server/serveractiveobject.h

```diff
@@ -51,11 +51,7 @@ struct PlayerHPChangeReason;
 class ServerActiveObject : public ActiveObject
 {
 public:
-	/*
-		NOTE: m_env can be NULL, but step() isn't called if it is.
-		Prototypes are used that way.
-	*/
-	ServerActiveObject(ServerEnvironment *env, v3f pos);
+	ServerActiveObject(ServerScripting *scriptIface, v3f pos);
```

Comment:
sorry but nope, attaching a SAO to the script class instead of the server env makes no sense.

The point here seems to be avoiding a server environment in the test but actually providing one shouldn't be *too* hard (does subclassing work?)

Reply:
It's the easiest way to make it work. No way I can provide that https://github.com/minetest/minetest/blob/0c4929f0252811d9715b96379973929053d32ef0/src/serverenvironment.cpp#L394-L410.

Reply:
Something worked out.